source/extensions/filters/http/decompressor/decompressor_filter.h

```diff
@@ -94,7 +94,7 @@ class DecompressorFilterConfig {
       Compression::Decompressor::DecompressorFactoryPtr decompressor_factory);
 
   Compression::Decompressor::DecompressorPtr makeDecompressor() {
-    return decompressor_factory_->createDecompressor();
+    return decompressor_factory_->createDecompressor(stats_prefix_ + "decompressor_library");
```

Comment:
I debated this back and forth because I am not sure if we would have a place where we would want each decompressor created by the same factory to have different strings.

Reply:
lets just do it the way you suggest. Having the string concatenation in the hot path seems like a bad price to pay for the niche ability to configure stats trees differently for individual decompressors.

Reply:
nvm all of the above. I now do understand what you mean. Let me know if what I have pushed is not it.

Reply:
That's definitely better. Though I was thinking about changing the meaning of `DecompressorFactory::statsPrefix()`. Instead of returning `.gzip` it could return the full root stats prefix since we create a factory per `DecompressionFilterConfig` anyway. 

Then `DecompressorFilterConfig::stats_prefix_` wouldn't be needed and `DecompressorFactory::createDecompressor()` would retain its current signature.

Reply:
I didn’t want to do that because I don’t want the filter stats to have the decompressor_library prefix. To me the public statsPrefix function is for a consumer to use. And the argument in createDeconoressor is what the consumer wants to ultimately root the decompressor at.

Reply:
Alright, fair enough.